Split/Network/Streaming/Synchronizer.swift

```diff
@@ -178,4 +199,43 @@ class DefaultSynchronizer: Synchronizer {
             task.stop()
         }
     }
+
+    private func filterSplitsInCache() {
+        let splitsStorage = splitStorageContainer.persistentSplitsStorage
+        let currentSplitsQueryString = splitsFilterQueryString
+        if currentSplitsQueryString == splitsStorage.getFilterQueryString() {
+            return
+        }
+
+        let filters = splitConfig.sync.filters
+        let namesToKeep = Set(filters.filter { $0.type == .byName }.flatMap { $0.values })
+        let prefixesToKeep = Set(filters.filter { $0.type == .byPrefix }.flatMap { $0.values })
+
+        let splitsInCache = splitsStorage.getAll()
+        var toDelete = [String]()
+
+        for split in splitsInCache {
+            guard let splitName = split.name else {
+                continue
+            }
+
+            if !namesToKeep.contains(splitName) {
```

Comment:
check this logic. 
I thinks it is wrongly deleting a split if it doesn't matches any name but matches a prefix.  

To delete a split, it shouldn't match neither with a name or a prefix. Something like:
```
if !namesToKeep.contains(splitName) && (!hasPrefix(for: splitName) || (let prefix = getPrefix(for: splitName) && !prefixesToKeep.contains(prefix)) ) {
    toDelete.append(splitName)
}
```

Reply:
Yup Doc, you're right! I've fixed the logic. Thank you!